### ISSUE TITLE
Add mdx support

### ISSUE THREAD

**Issue by yury** [facebook/flow#6873]
It would be awesome if we can type check mdx files.

See  https://github.com/mdx-js/mdx/issues/232

If you can point and direction how it could be done, I will try to implement it :)

**Comment by SidOfc** [comment]
Hello!

I got here from mdx-js/mdx#232 as well :)

I'm using flow with NextJS and am currently working on a side project to implement a small blog.
As usual, a wonderful solution already exists in the form of [mdx-js](https://github.com/mdx-js/mdx).
Unfortunately, Flow doesn't understand when I `import` something from a file ending in `.mdx`:

![image](https://user-images.githubusercontent.com/3225058/57875774-fdd7c980-7813-11e9-9d9f-0ae87ccfe1b8.png)

**./blog/codi-vim-fullscreen-buffer-workflow.mdx**
```mdx
export const meta = {
  title: 'Codi.vim + fullscreen buffer workflow',
  author: 'Sidney Liebrand',
  date: '27-08-2018',
  excerpt: 'short description of post',
  words: 874
};

# Codi.vim + fullscreen buffer workflow

... snipped ...
```

I would love to see this feature supported as well since I really like Flow, and NextJS + mdx-js seems like a good combo to apply some _Flow-fu_ to.

If there's anything I can do and some guidance is available when needed, I wouldn't mind looking at how this can be implemented either :)

**Comment by goodmind** [comment]
@SidOfc do you want imports to work, or typechecking too?

**Comment by SidOfc** [comment]
@goodmind I'd be fine with just imports personally, if type checking could be added without massive amounts of work that would be (very) nice to have. Though, if imports work fine yet type checking seems to fail then it could look like another bug in flow rather than an unsupported feature.

I must say that looking at the code base, I am humbled by inexperience while _trying_ to read the syntax and grasp the structure. So unsure if I can actually be of any use (at least any time soon, rather than being a concrete block stuck to your leg) 😅 

Have to hand it to you, 🔥fast response time 👍

**Comment by goodmind** [comment]
You can enable imports with `module.file_ext` option in `.flowconfig`

**Comment by SidOfc** [comment]
Thank you for supplying these great alternatives!

I failed to see the [flow config options docs](https://flow.org/en/docs/config/options) which indeed perfectly describes how to add additional extensions using [`module.file_ext`](https://flow.org/en/docs/config/options/#toc-module-file-ext-string), my bad there.

Based on your answer I tried to Google for `.js.flow files` and `flow ".js.flow" files`, both [results](https://www.google.com/search?q=.js.flow+files&oq=.js.flow+files&aqs=chrome..69i57j0l6.2655j0j4&sourceid=chrome&ie=UTF-8) are more or less identical. Two issues about "Document .js.flow files" come up as first results. It seems like facebook/flow#3966 was closed in favor of still-open facebook/flow#3609. 

Are there any plans (facebook/flow#3609 has been open for quite some time) to document this still? I think it will be nice for folks looking for documentation on this to be able to find it on the official website rather than trying to figure out what is correct from some GH issues / SO answers.

That is actually how I ended up posting here, I didn't know this was even possible. I am not a native English speaker though, so maybe specific wording passed by me in the docs, would love to be pointed to them if so :)

Aside from that my issues has been solved by examples, thank you once again for the quick support and explanation!
